### ISSUE TITLE
got an error on installing lightgbm on linux

### ISSUE THREAD

**Issue by franciszmy** [microsoft/LightGBM#2057]
I have used lightgbm by using rstudio on centOS before, but for some reasons, yesterday I deleted my lightgbm package on rstudio server and when I tried to reinstall it later, it posted an error which I can't solve. 

The error is

** testing if installed package can be loaded
Error in UseMethod("conditionCall") : 
  no applicable method for 'conditionCall' applied to an object of class "character"
Calls: <Anonymous> ... tryCatchList -> tryCatchOne -> <Anonymous> -> conditionCall
Execution halted
ERROR: loading failed
* removing ‘/home/xxxx/R/x86_64-pc-linux-gnu-library/3.4/lightgbm’

this error seems a base function in R, but I don't know why something is wrong with this base function.

Anyone could help me? Thanks so much in advance!!!

**Comment by jameslamb** [comment]
Hi @franciszmy , we're here to help!

Can you please share the following?

1. The steps you used to remove `lightgbm`
2. The steps you used to reinstall it
3. The output for running `sessionInfo()`

**Comment by franciszmy** [comment]
thanks james, here are some information about what you mentioned above.

1.I just removed package from rstudio GUI. No other extra moves
2.I have lightgbm 2.0.1.0 package. I reinstalled it by using install.package in R
3. sessionInfo()
R version 3.4.4 (2018-03-15)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS release 6.10 (Final)

Matrix products: default
BLAS: /opt/rinstall/R-3.4.4/lib/libRblas.so
LAPACK: /opt/rinstall/R-3.4.4/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] R6_2.4.0      xgboost_0.6-4

loaded via a namespace (and not attached):
[1] compiler_3.4.2      magrittr_1.5        Matrix_1.2-11       tools_3.4.2         stringi_1.1.6      
[6] grid_3.4.2          data.table_1.10.4-3 lattice_0.20-35    

Besides, the reason why I delete my lightgbm package is because I cannot use lightgbm package that I used to use successfully, the reason why I cannot use lightgbm is because I find later that the R is updated to 3.5 , so I reinstall R with 3.4 version.


below is the details of installation information:

Installing package into ‘/home/xxxx/R/x86_64-pc-linux-gnu-library/3.4’
(as ‘lib’ is unspecified)
* installing *source* package ‘lightgbm’ ...
** libs
installing via 'install.libs.R' to /home/xxxx/R/x86_64-pc-linux-gnu-library/3.4/lightgbm
-- The C compiler identification is GNU 4.4.7
-- The CXX compiler identification is GNU 4.8.5
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Try OpenMP C flag = [-fopenmp]
-- Performing Test OpenMP_FLAG_DETECTED
-- Performing Test OpenMP_FLAG_DETECTED - Success
-- Try OpenMP CXX flag = [-fopenmp]
-- Performing Test OpenMP_FLAG_DETECTED
-- Performing Test OpenMP_FLAG_DETECTED - Success
-- Found OpenMP: -fopenmp  
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/Rtmpi544bM/R.INSTALL565976cfeab4/lightgbm/src/build
Scanning dependencies of target _lightgbm
[  3%] [  7%] [ 17%] [ 17%] [ 17%] [ 21%] [ 25%] [ 28%] [ 32%] [ 39%] [ 39%] [ 42%] [ 46%] [ 50%] [ 57%] [ 57%] Building CXX object CMakeFiles/_lightgbm.dir/src/c_api.cpp.o
Building CXX object CMakeFiles/_lightgbm.dir/src/application/application.cpp.o
Building CXX object CMakeFiles/_lightgbm.dir/src/boosting/gbdt_model_text.cpp.o
Building CXX object CMakeFiles/_lightgbm.dir/src/lightgbm_R.cpp.o
[ 60%] Building CXX object CMakeFiles/_lightgbm.dir/src/boosting/boosting.cpp.o
Building CXX object CMakeFiles/_lightgbm.dir/src/boosting/prediction_early_stop.cpp.o
[ 64%] [ 67%] Building CXX object CMakeFiles/_lightgbm.dir/src/io/dataset.cpp.o
Building CXX object CMakeFiles/_lightgbm.dir/src/io/metadata.cpp.o
Building CXX object CMakeFiles/_lightgbm.dir/src/boosting/gbdt.cpp.o
[ 71%] Building CXX object CMakeFiles/_lightgbm.dir/src/io/dataset_loader.cpp.o
Building CXX object CMakeFiles/_lightgbm.dir/src/boosting/gbdt_prediction.cpp.o
Building CXX object CMakeFiles/_lightgbm.dir/src/io/bin.cpp.o
Building CXX object CMakeFiles/_lightgbm.dir/src/io/config.cpp.o
Building CXX object CMakeFiles/_lightgbm.dir/src/io/parser.cpp.o
Building CXX object CMakeFiles/_lightgbm.dir/src/io/tree.cpp.o
Building CXX object CMakeFiles/_lightgbm.dir/src/metric/dcg_calculator.cpp.o
[ 75%] [ 78%] [ 82%] Building CXX object CMakeFiles/_lightgbm.dir/src/metric/metric.cpp.o
[ 85%] [ 89%] Building CXX object CMakeFiles/_lightgbm.dir/src/objective/objective_function.cpp.o
[ 92%] Building CXX object CMakeFiles/_lightgbm.dir/src/network/network.cpp.o
[ 96%] Building CXX object CMakeFiles/_lightgbm.dir/src/network/linkers_socket.cpp.o
[100%] Building CXX object CMakeFiles/_lightgbm.dir/src/network/linker_topo.cpp.o
Building CXX object CMakeFiles/_lightgbm.dir/src/network/linkers_mpi.cpp.o
Building CXX object CMakeFiles/_lightgbm.dir/src/treelearner/tree_learner.cpp.o
Building CXX object CMakeFiles/_lightgbm.dir/src/treelearner/data_parallel_tree_learner.cpp.o
Building CXX object CMakeFiles/_lightgbm.dir/src/treelearner/gpu_tree_learner.cpp.o
Building CXX object CMakeFiles/_lightgbm.dir/src/treelearner/serial_tree_learner.cpp.o
Building CXX object CMakeFiles/_lightgbm.dir/src/treelearner/voting_parallel_tree_learner.cpp.o
Building CXX object CMakeFiles/_lightgbm.dir/src/treelearner/feature_parallel_tree_learner.cpp.o
Linking CXX shared library ../lib_lightgbm.so
[100%] Built target _lightgbm
Found library file: /tmp/Rtmpi544bM/R.INSTALL565976cfeab4/lightgbm/src/lib_lightgbm.so to move to /home/xxxx/R/x86_64-pc-linux-gnu-library/3.4/lightgbm/libs
** R
** data
** demo
** preparing package for lazy loading
** help
*** installing help indices
  converting help for package ‘lightgbm’
    finding HTML links ... done
    agaricus.test                           html  
    agaricus.train                          html  
    bank                                    html  
    dim                                     html  
    dimnames.lgb.Dataset                    html  
    getinfo                                 html  
    lgb.Dataset                             html  
    lgb.Dataset.construct                   html  
    lgb.Dataset.create.valid                html  
    lgb.Dataset.save                        html  
    lgb.Dataset.set.categorical             html  
    lgb.Dataset.set.reference               html  
    lgb.dump                                html  
    lgb.get.eval.result                     html  
    lgb.importance                          html  
    lgb.interprete                          html  
    lgb.load                                html  
    lgb.model.dt.tree                       html  
    lgb.plot.importance                     html  
    lgb.plot.interpretation                 html  
    lgb.prepare                             html  
    lgb.prepare2                            html  
    lgb.prepare_rules                       html  
    lgb.prepare_rules2                      html  
    lgb.save                                html  
    lgb.train                               html  
    lgb.unloader                            html  
    predict.lgb.Booster                     html  
    readRDS.lgb.Booster                     html  
    saveRDS.lgb.Booster                     html  
    setinfo                                 html  
    slice                                   html  
** building package indices
** testing if installed package can be loaded
Error in UseMethod("conditionCall") : 
  no applicable method for 'conditionCall' applied to an object of class "character"
Calls: <Anonymous> ... tryCatchList -> tryCatchOne -> <Anonymous> -> conditionCall
Execution halted
ERROR: loading failed
* removing ‘/home/xxxx/R/x86_64-pc-linux-gnu-library/3.4/lightgbm’
Warning in install.packages :
  installation of package ‘/home/xxxx/lightgbm_2.0.10.tar.gz’ had non-zero exit status 


thank you so much!

**Comment by jameslamb** [comment]
Hmmm I'm still trying to figure this one out.

That error message is a developer error _somewhere_ in the code used to load packages. It indicates that some R code is relying on S3 dispatching but got a type of data with no registered method.

You're saying that you _downgraded_ from R 3.5.x to R 3.4.x? Why did you have to do that? My best guess right now is that something was left behind on your system and that R or `cmake` thinks something is cached that actually needs to be recompiled.

R 3.5 was a significant departure from 3.4 (https://cran.r-project.org/doc/manuals/r-devel/NEWS.html) so it's possible for a package that worked on 3.5 to not be installable on 3.4.

The R side of our CI needs some work, so I cannot say with certainty what versions of R we support. I'll try to reproduce your environment and see if I can get the same error!

While I try to do that, could you manually remove `/home/xxxx/R/x86_64-pc-linux-gnu-library/3.4/lightgbm` and try re-installing?

One other thing...is there still a `/home/xxxx/R/x86_64-pc-linux-gnu-library/3.5` left behind? When I've used RStudio Server with a shared file system in the past, sometimes I've needed to remove other `R/` directories left behind when changing versions.

**Comment by StrikerRUS** [comment]
ping @franciszmy

**Comment by StrikerRUS** [comment]
@franciszmy Have you managed to try @jameslamb's advices about remaining folders?

**Comment by Laurae2** [comment]
@jameslamb @franciszmy Just had the same issue last week when switching to R 3.6 (except it's from R 3.5.3) on another package. RStudio Server is not fun to use when switching R versions.

In that case, it was better to nuke (delete) the hidden `.rstudio` folder to fully reset RStudio Server. Note that doing it will delete also all customizations done in RStudio Server (which have to be re-done manually).

**Comment by jameslamb** [comment]
😬 I feel that we have a set of strange R 3.6.0 issues coming our way 😬 

@franciszmy could you try mine and @Laurae2 's suggestions and let us know if the issue gets resolved?

**Comment by StrikerRUS** [comment]
Closing due to inactivity.
